rama-net/src/client/pool.rs

```diff
@@ -231,41 +275,33 @@ where
 
         let mut storage = self.storage.lock();
 
-        let idx = storage.iter().position(|stored| &stored.id == id);
-        let pooled_conn = idx.and_then(|idx| storage.remove(idx));
-
-        if let (Some(idx), Some(pooled_conn)) = (idx, pooled_conn) {
-            trace!("fifo connection pool: connection #{idx} found for given id {id:?}");
-            let mut timeout_triggered = false;
-            if let Some(timeout) = self.idle_timeout {
-                if pooled_conn.last_used.elapsed() > timeout {
-                    timeout_triggered = true;
-                }
-            }
-
-            // Since we always insert in the beginning we can assume if this connection has been idle for too long
-            // all connections older then this one should also be evicted
-            if timeout_triggered {
+        if let Some(timeout) = self.idle_timeout {
+            let now = Instant::now();
+            let idx = storage.partition_point(|conn| now.duration_since(conn.last_used) <= timeout);
+            if idx < storage.len() {
                 trace!(
-                    "fifo connection pool: idle timeout was triggered, dropping this connection (w/ start index {idx:?}) and all older ones"
+                    "lru connection pool: idle timeout was triggered, dropping connections with index {idx:?} and later"
                 );
                 storage.drain(idx..);
-            } else {
-                return Ok(ConnectionResult::Connection(LeasedConnection {
-                    active_slot,
-                    pooled_conn: Some(pooled_conn),
-                    returner: self.returner.clone(),
-                    failed: false.into(),
-                }));
             }
         }
 
+        if let Some((idx, pooled_conn)) = self.reuse_strategy.get_conn(&mut storage, id) {
+            trace!("lru connection pool: connection #{idx} found for given id {id:?}");
+            return Ok(ConnectionResult::Connection(LeasedConnection {
+                active_slot,
+                pooled_conn: Some(pooled_conn),
+                returner: self.returner.clone(),
+                failed: false.into(),
```

Comment:
I prefer you use notations like `AtomicBool::new` here so it's clear from reading this one line what is getting created here (without having to have an IDE to do so)